scipy/optimize/optimize.py

```diff
@@ -676,6 +676,19 @@ def _minimize_neldermead(func, x0, args=(), callback=None,
 
     x0 = asfarray(x0).flatten()
 
+    if bounds is None:
+        lower_bound, upper_bound = np.array((-np.inf, np.inf))
+    else:
+        lower_bound, upper_bound = bounds.lb, bounds.ub
+        # check bounds
+        if (lower_bound > upper_bound).any():
```

Comment:
Could you add a test to check what would happen to the function if there were equal bounds for one or more of the parameters? This causes issues with gradient based minimisers.

Reply:
I added it. It seems that it works well. In `test_equal_all_bounds`, the result is same as the equal bounds. In `test_equal_one_bounds`, the one of result is on the equal bound, the another is optimal (4**2 = 16) in rosen problem.